examples/audio-speech-stream/src/main.rs

```diff
@@ -0,0 +1,45 @@
+use async_openai::{
+    types::{CreateSpeechRequestArgs, SpeechResponseFormat, SpeechModel, Voice},
+    Client,
+};
+use std::error::Error;
+use std::fs::File;
+use std::io::Write;
+use futures::{future, StreamExt};
+use bytes::{Buf, Bytes, BytesMut, BufMut};
+use bytes::buf::Reader;
+
+#[tokio::main]
+async fn main() -> Result<(), Box<dyn Error>> {
+    let client = Client::new();
+
+    let request = CreateSpeechRequestArgs::default()
+        .input("Today is a wonderful day to stream something people love!")
+        .voice(Voice::Alloy)
+        .model(SpeechModel::Tts1)
+        .response_format(SpeechResponseFormat::Mp3)
+        .build()?;
+
+    let mut stream = client.audio().speech_stream(request).await?;
+
+    let mut file = File::create("./speech_stream.mp3")?; 
+    
+    stream.for_each(|item| {
+        match item {
+            Ok(response) => {
+                let res_size = file.write(&response.bytes);
```

Comment:
This example is not working as intended. If I dont provide API key I would expect error to log, but instead it writes the error json to mp3 file.

Reply:
The example does work when provided the API key, but i"m still uncertain how without `stream=True` ?

Reply:
Ah - this is my bad. I had assumed that a different status code from 200 would go in the Result::Err branch... but now I see that the conversion logic is part of the `execute_raw` function. I will move that code around so I can reuse it and ensure that I do some error tests as well.